data/virt_autotest/virt_logs_collector.sh

```diff
@@ -414,21 +414,54 @@ guest_hash_index=0
 guest_current=""
 dhcpd_lease_file="/var/lib/dhcp/db/dhcpd.leases"
 
+#Install necessary packages
+echo -e "Install necessary packages. zypper install -y sshpass nmap xmlstarlet" | tee -a ${virt_logs_collecor_log}
+zypper install -y sshpass nmap xmlstarlet | tee -a ${virt_logs_collecor_log}
+
+#Establish reachable networks and hosts database on host
+subnets_in_route=`ip route show all | awk '{print $1}' | grep -v default`
+subnets_scan_results=""
+subnets_scan_index=0
+echo -e "Subnets ${subnets_in_route[@]} are reachable on host judging by ip route show all" | tee -a ${virt_logs_collecor_log}
+echo -e "Establishing reachable hosts in subnets ${subnets_in_route[@]} database on host" | tee -a ${virt_logs_collecor_log}
+for single_subnet in ${subnets_in_route[@]};do
+    single_subnet_transformed=${single_subnet//./_}
+    single_subnet_transformed=${single_subnet_transformed/\//_}
+    scan_timestamp=`date "+%F-%H-%M-%S"`
+    mkdir -p "${virt_logs_folder}/nmap_subnets_scan_results"
+    single_subnet_scan_results=${virt_logs_folder}'/nmap_subnets_scan_results/nmap_scan_'${single_subnet_transformed}'_'${scan_timestamp}
+    subnets_scan_results[${subnets_scan_index}]=${single_subnet_scan_results}
+    echo -e "nmap -sn $single_subnet -oX $single_subnet_scan_results" | tee -a ${virt_logs_collecor_log}
```

Comment:
about the 192.168.123.0/24 subnet, because it is the host/guests dhcp network for our testing only, so the log is useful. but about the 10.67.128.0/21 subnet, is it necessary? Should we scan all the hosts in this team/site-wide network? Is it secure to upload the logs to public?

Reply:
Do you mean uploaded to openQA or here(on Github) ?
openQA needs SUSE account, corresponding authorization and connect to R&D network, so I do not think this poses any threat to the internal. We also put developing builds on openQA even before going public, are they for free use ? Of course not. And a failure job or its worker log also shows detailed ipmi/host connection information, I think this already answers what you are concerned about.
If you think I should not put such log on Github, I can remove it.

Reply:
I think "checking private checkbox" is your choice on Bugzilla if this really concerns you.

Reply:
Could you explain why you insist on publishing the hosts information of this subnet (10.162.2.xx in OSD or 10.167.129.xx in apacii)? Where are these logs useful? I can't imagine where I need these logs in my tests.